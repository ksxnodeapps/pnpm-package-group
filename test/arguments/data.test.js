'use strict'
const {resolve} = require('path')
const {readFileSync} = require('fs')

const getText = basename =>
  readFileSync(resolve(__dirname, '../../lib/arguments', basename), 'utf8')

const yaml = require('js-yaml').safeLoad(getText('data.yaml'))
const json = JSON.parse(getText('data.json'))

describe('data.yaml and data.json', () => {
  it('stays unchanged', () => {
    expect({yaml, json}).toMatchSnapshot()
  })

  it('should be equal', () => {
    expect(yaml).toEqual(json)
  })
})
