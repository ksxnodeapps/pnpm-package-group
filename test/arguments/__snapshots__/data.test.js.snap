// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data.yaml and data.json should stay unchanged 1`] = `
Object {
  "json": Object {
    "env": "PNPM_PKG_GROUP",
    "example": Array [
      Object {
        "cmd": "$0 packages.yaml",
        "desc": "Basic use",
      },
      Object {
        "cmd": "$0 --local=local-packages packages.yaml",
        "desc": "Specify target directory",
      },
      Object {
        "cmd": "$0 packages.yaml/foo/bar",
        "desc": "Install only group foo.bar in packages.yaml",
      },
      Object {
        "cmd": "$0 /abc/def/packages.yaml/foo/bar",
        "desc": "Combine filesystem path and json path",
      },
      Object {
        "cmd": "$0 file1.yaml/foo/bar1 file1.yaml/foo/bar2 file2.yaml/foo/bar1 file2.yaml/foo/bar2",
        "desc": "Filter files and groups",
      },
      Object {
        "cmd": "$0 *.yaml/foo/{bar1,bar2}",
        "desc": "Using blob pattern (UNIX Shell only)",
      },
    ],
    "options": Object {
      "global-syntax": Object {
        "alias": "G",
        "default": "{cmd} install --global {list}",
        "describe": "Syntax to install packages globally",
        "type": "string",
      },
      "help": Object {
        "alias": "h",
      },
      "local": Object {
        "alias": "l",
        "default": "",
        "describe": "Directory contains local packages",
        "type": "string",
      },
      "local-syntax": Object {
        "alias": "L",
        "default": "{cmd} install --save {list}",
        "describe": "Syntax to install packages locally",
        "type": "string",
      },
      "pnpm": Object {
        "default": "pnpm",
        "describe": "Program to call",
        "type": "string",
      },
      "quiet-pnpm": Object {
        "default": false,
        "describe": "Don't print pnpm output",
        "type": "boolean",
      },
      "shell": Object {
        "default": true,
        "describe": "Whether shell is used",
        "type": "boolean",
      },
    },
    "usage": "$ $0 [options] [arguments]",
  },
  "yaml": Object {
    "env": "PNPM_PKG_GROUP",
    "example": Array [
      Object {
        "cmd": "$0 packages.yaml",
        "desc": "Basic use",
      },
      Object {
        "cmd": "$0 --local=local-packages packages.yaml",
        "desc": "Specify target directory",
      },
      Object {
        "cmd": "$0 packages.yaml/foo/bar",
        "desc": "Install only group foo.bar in packages.yaml",
      },
      Object {
        "cmd": "$0 /abc/def/packages.yaml/foo/bar",
        "desc": "Combine filesystem path and json path",
      },
      Object {
        "cmd": "$0 file1.yaml/foo/bar1 file1.yaml/foo/bar2 file2.yaml/foo/bar1 file2.yaml/foo/bar2",
        "desc": "Filter files and groups",
      },
      Object {
        "cmd": "$0 *.yaml/foo/{bar1,bar2}",
        "desc": "Using blob pattern (UNIX Shell only)",
      },
    ],
    "options": Object {
      "global-syntax": Object {
        "alias": "G",
        "default": "{cmd} install --global {list}",
        "describe": "Syntax to install packages globally",
        "type": "string",
      },
      "help": Object {
        "alias": "h",
      },
      "local": Object {
        "alias": "l",
        "default": "",
        "describe": "Directory contains local packages",
        "type": "string",
      },
      "local-syntax": Object {
        "alias": "L",
        "default": "{cmd} install --save {list}",
        "describe": "Syntax to install packages locally",
        "type": "string",
      },
      "pnpm": Object {
        "default": "pnpm",
        "describe": "Program to call",
        "type": "string",
      },
      "quiet-pnpm": Object {
        "default": false,
        "describe": "Don't print pnpm output",
        "type": "boolean",
      },
      "shell": Object {
        "default": true,
        "describe": "Whether shell is used",
        "type": "boolean",
      },
    },
    "usage": "$ $0 [options] [arguments]",
  },
}
`;
